### PR TITLE
Core: Fix - Region events

### PR DESCRIPTION
**Pull request by cjmconie**
Create symmetric events: will and did

### CODE REVIEW

File: BlockV/Core/Data Pool/Regions/Region+Notifications.swift [hunk 1]
```diff
@@ -16,15 +16,29 @@ public enum RegionEvent: String {
 
     /// Triggered when any data in the region changes. This also indicates that there is no longer an error.
     case updated = "region.updated"
+    
+    // will change
 
     /// Triggered when an object is added.
-    case objectAdded = "region.object.added"
+    case willAddObject = "region.object.will_adde"
```
Comment (MalcolmnDEV) [inline BlockV/Core/Data Pool/Regions/Region+Notifications.swift]: missing a 'd at the end'

File: BlockV/Core/Data Pool/Regions/BLOCKvRegion.swift [hunk 2]
```diff
@@ -386,11 +386,27 @@ class BLOCKvRegion: Region {
...
+            self.emit(.willRemoveObject, userInfo: ["id": object.id])
         }
 
     }
+    
+    /// Called when an object has been removed.
+    override func did(remove object: DataObject) {
+        
+        // notify parent as well
+        guard let parentID = (object.data?["vAtom::vAtomType"] as? [String: Any])?["parent_id"] as? String else {
```
Comment (MalcolmnDEV) [inline BlockV/Core/Data Pool/Regions/BLOCKvRegion.swift]: can we not split this into 2 lines?

File: BlockV/Core/Data Pool/Regions/Region+Notifications.swift (hunk 1 above)
Comment (MalcolmnDEV) [reply BlockV/Core/Data Pool/Regions/Region+Notifications.swift]: or remove 'e'

Comment (cjmconie) [reply BlockV/Core/Data Pool/Regions/Region+Notifications.swift]: Fixed

File: BlockV/Core/Data Pool/Regions/BLOCKvRegion.swift (hunk 2 above)
Comment (cjmconie) [reply BlockV/Core/Data Pool/Regions/BLOCKvRegion.swift]: It's within the line_length limit, and appears elsewhere in the code in a very similar pattern so I'd prefer not to change it right now.

Comment (cjmconie) [reply BlockV/Core/Data Pool/Regions/BLOCKvRegion.swift]: Updated.

**Pull request merged**